lib/graide/gdx.py

```diff
@@ -83,8 +83,8 @@ def __init__(self, e, relbase) :
         self.srcline = int(e.get('atLine')) - 1
         self.pretty = e.get('prettyPrint')
         slots = map(lambda x: int(x.get('slotIndex')), e.findall('rhsSlot'))
-        if len(slots) :
-            d = slots[0]
+        if any(slots) :
+            d = next(slots)
             self.slots = map(lambda x : x - d, slots)
         else :
-            self.slots = []
+            self.slots = {}
```

Comment:
why change from empty list to empty dict here?

Reply:
A dict is more aligned with what map() returns.

Reply:
I'm pretty sure map on python2 returns a list and on python3 an iterator